near-sdk/src/json_types/public_key.rs

```diff
@@ -77,19 +87,29 @@ impl Base58PublicKey {
                 kind: ParsePublicKeyErrorKind::InvalidLength(data.len()),
             });
         }
+        let mut bytes = Vec::with_capacity(1 + expected_length);
+        bytes.push(curve as u8);
+        bytes.extend(data);
 
-        Ok(Self { curve, data })
+        Ok(Self { data: bytes })
     }
 
     /// Get info about the CurveType for this public key
     pub fn curve_type(&self) -> CurveType {
-        self.curve
+        CurveType::from_u8(self.data[0]).unwrap()
```

Comment:
Same, better to avoid unwraps on results which shouldn't be possible to avoid formatting related code bloat.